### PR TITLE
feat(chore): PostgresEngine class methods

### PR DESCRIPTION
**Pull request by Itzaprado**
[ ] Adding from_engine, from_engine_args
[ ] Adding init_vectorstore_table
[ ] Adding init_chat_history_table
[ ] Adding testing for each method

### CODE REVIEW

File: libs/langchain-google-cloud-sql-pg/src/engine.ts [hunk 1]
```diff
@@ -10,6 +10,44 @@ export interface PostgresEngineArgs {
   iamAccountEmail?: string,
 }
 
+export interface VectorStoreTableArgs {
+  tableName: string,
+  vectorSize: number,
```
Comment (averikitsch) [inline libs/langchain-google-cloud-sql-pg/src/engine.ts]: Similar to the Engine initialization, can we have the tableName and vectorSize as required args and separate from the Args dict of optional ars?

File: libs/langchain-google-cloud-sql-pg/src/engine.ts [hunk 2]
```diff
@@ -96,12 +134,125 @@ class PostgresEngine {
...
+  static async from_engine_args(url: string | knex.Knex.StaticConnectionConfig) {
+
+    const driver = 'postgresql+asyncpg';
+
+    // if((typeof url === "string") && !url.startsWith(driver) || (url instanceof URL && url.protocol !== driver)){
+    if((typeof url === "string") && !url.startsWith(driver)){
+      throw "Driver must be type 'postgresql+asyncpg'"
+    }
+
+    const dbConfig: knex.Knex.Config<any> = {
```
Comment (averikitsch) [inline libs/langchain-google-cloud-sql-pg/src/engine.ts:160]: Can we also support the user passing in a map of configs to extend the dbConfig object?

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: libs/langchain-google-cloud-sql-pg/src/engine.ts (hunk 2 above)
Comment (Itzaprado) [reply libs/langchain-google-cloud-sql-pg/src/engine.ts:160]: Could you be more specific? The dbConfig here uses the URL to connect to the database, and this already contains the database user, I don't think it's necessary to add the user to the configuration.

File: libs/langchain-google-cloud-sql-pg/src/tests/engine.test.ts [hunk 3]
```diff
@@ -82,5 +89,140 @@ describe("PostgresEngine Instance creation", () => {
...
+
+    async function createInstance() {
+      PEInstance = await PostgresEngine.from_engine_args(url);
+    }
+
+    await expect(createInstance).rejects.toBe("Driver must be type 'postgresql+asyncpg'");
+  });
+
+  test('should create a PostgresEngine Instance through from_engine_args using a URL', async () => {
+    const url = process.env.PG_URL ?? "";
```
Comment (averikitsch) [inline libs/langchain-google-cloud-sql-pg/src/tests/engine.test.ts]: Can you construct this like "postgresql+asyncpg://{USER}:{PASSWORD}@{HOST}:5432/{DATABASE_NAME}", so the only new env var is the Host?

File: libs/langchain-google-cloud-sql-pg/src/engine.ts [hunk 4]
```diff
@@ -96,12 +132,127 @@ class PostgresEngine {
...
+      query += `,\n ${column.name} ${column.dataType} ${nullable}`;
+    }
+
+    if(storeMetadata) {
+      query += `,\n${metadataJsonColumn} JSON`;
+    }
+
+    query += `\n);`
+
+    console.log(query)
```
Comment (averikitsch) [inline libs/langchain-google-cloud-sql-pg/src/engine.ts]: remove

**Review by averikitsch — APPROVED** [review]

**Pull request merged**